analyze_logs/src/analyzer.py

```diff
@@ -58,8 +66,7 @@ def fullAnalysis(self):
             e = sys.exc_info()[0]
             self.log.error("Could not process: " + event.message_text + " because %s" % e)
             traceback.print_exc(file=sys.stdout)
-      else:
-        break 
+      time.sleep(self.sleep_time)   
```

Comment:
I believe the way it used to work was to continue pulling items off the queue until there are none left, at which point it breaks and exits. I changed it so you have the option not to break (with the keep running flag), and then if you haven't exit, the sleep time stops it from spinning until there is something on the queue again. The default behaviour should be unchanged.